examples/gallery/database_module.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+"""
+=========================================
+The Database Module
+=========================================
+
+In this example we demonstrate some of the features of the Database module.
+"""
+
+##############################################################################
+# :ref:`database_guide` gives a detailed and comprehensive guide on how
+# to use the Database package.
+
+##############################################################################
+# Let's start by importing the necessary modules.
+
+from sunpy.database import Database
+from sunpy.database.tables import display_entries
+from sunpy.net import vso
+
+##############################################################################
+# First we need to instantiate a `~sunpy.database.Database` object. Note that
+# the argument can be any URL supported by SQLAlchemy.
+
+database = Database('sqlite:///sunpydata.sqlite')
+
+##############################################################################
+# Adding Entries from a VSO query
+
+vso_query_result = vso.VSOClient().query(
+	vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
+	vso.attrs.Instrument('AIA'))
+database.add_from_vso_query_result(vso_query_result)
+
+##############################################################################
+# Now to see what is in the database, we use the display_entries function to
+# display the desired parameters in a neat way.
+
+print(display_entries(database,
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Downloading data. We can download the required files and store them so
+# that they can be easily accessed for future use. An optional argument path
+# can be passed to specify the download location.
+
+database.download(vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
+	vso.attrs.Instrument('AIA'))
+
+##############################################################################
+# After downloading, The number of entries that will be added depends on the
+# total number of FITS headers.
+
+print(display_entries(database,
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Clever Fetching.
+# The fetch() method queries the database if the given query
+# has already been used once to add entries using the download() method.
+# Otherwise, the given query is used to download and add new data via the
+# download() method. This means that if you are not sure whether the required
+# data already exists in the database, use fetch() to save bandwidth.
+
+print("Number of entries in database : " + str(len(database)))
+
+##############################################################################
+
+entries = database.fetch(vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
```

Comment:
on a higher level, I am not sure I like the distinction between the `download` and `fetch` methods. It get's confusing to me, so I can't imaging that it's clear to end users.